Update to tiled fix files, UGWD v1, NOAH-MP

WalterKolczynski-NOAA opened the pull request:
Updates model to use tiled fix files. Fix directory is updated to the
fix_NEW location.

Adds the ability to use UGWD v1. Since this capability is tied to the
CCPP suite used, the sutie definition file is grepped to determine whether
UGWD is active. Otherwise, gwd_opt 1 is used. Either way, the appropriate
namelist settings are added to input.nml. For v1, the necessary fix files
are also linked to the run directory. If additional options are supported
in the future, there will need to be more sophisicated parsing.

Adds the ability to use Noah-MP. Like UGWD, this is dictated by the CCPP
suite used, so the suite definition file is grepped to determine whether
to use Noah (lsm=1) or Noah-MP (lsm=2).

Additional CCPP suites are added to the build to allow for the new options.

The two non-fractional coupled cases are updated to use a CCPP suite using
both UGWD v1 and Noah-MP. The two aerosol cases are updated to a suite using
UGWD v1 (there does not appear to be an atm-only suite that has both).

There is also a minor change to the UFS build script to remove any existing
UFS build directory. This prevents problems when attempting to build a
different app after one has already been built.

Closes: #331, #346

WalterKolczynski-NOAA committed:
Update to tiled fix files, UGWD v1, NOAH-MP

Updates model to use tiled fix files. Fix directory is updated to the
fix_NEW location.

Adds the ability to use UGWD v1. Since this capability is tied to the
CCPP suite used, the sutie definition file is grepped to determine whether
UGWD is active. Otherwise, gwd_opt 1 is used. Either way, the appropriate
namelist settings are added to input.nml. For v1, the necessary fix files
are also linked to the run directory. If additional options are supported
in the future, there will need to be more sophisicated parsing.

Adds the ability to use Noah-MP. Like UGWD, this is dictated by the CCPP
suite used, so the suite definition file is grepped to determine whether
to use Noah (lsm=1) or Noah-MP (lsm=2).

Additional CCPP suites are added to allow for the new options.

The two non-fractional coupled cases are updated to use a CCPP suite using
both UGWD v1 and Noah-MP. The two aerosol cases are updated to a suite using
UGWD v1 (there does not appear to be an atm-only suite that has both).

There is also a minor change to the UFS build script to remove any existing
UFS build directory. This prevents problems when attempting to build a
different app after one has already been built.

Closes: #331, #346

WalterKolczynski-NOAA commented:
Confirmed to work for coupled_wave, coupled, and firex. Also tested firex with the old CCPP suite to confirm the old settings still work.

rmontuoro commented on ush/forecast_postdet.sh:
```diff
@@ -198,21 +198,46 @@ FV3_GFS_postdet(){
-	done
 
-	# GFS standard input data
+	_suite_file=$HOMEgfs/sorc/ufs_coupled.fd/FV3/ccpp/suites/suite_${CCPP_SUITE}.xml
```
Should this statement be guarded? What happens if suite_${CCPP_SUITE}.xml does not exist?

WalterKolczynski-NOAA replied:
Other portions of the same script already use it, so I assumed it was guaranteed to be set. That does not appear to be the case after all. I will move the default from parsing_namelists_FV3.sh up somewhere to make sure it is set.

WalterKolczynski-NOAA replied:
And yes, should also check that the file exists.

WalterKolczynski-NOAA replied:
Fixed now

rmontuoro reviewed (approved):
Confirming that these changes work with aerosols.
One minor comment to be addressed.

JessicaMeixner-NOAA reviewed (approved):
The tiled fix files look good to me and the fix files for uGWD also seem to be moved over as they're in the rt.sh.  I was able to create a successful run on hera.

WalterKolczynski-NOAA committed:
Ensure CCPP_SUITE is set and suite file exists

Previously there was no guarantee that CCPP_SUITE was set before it
was used in forecast_postdet.sh. The suite how has a default value if
it is not set. A default later in the execution chain is removed, as
CCPP_SUITE is now guaranteed to be set beforehand. The default is also
updated to FV3_GFS_v16.

There is also now a check to ensure the suite file is present inside
the UFS directory before trying to determine settings from it.

Refs: #346
